app/src/main/java/ip/team13/petowner/ui/login/LoginViewModel.kt

```diff
@@ -1,6 +1,23 @@
 package ip.team13.petowner.ui.login
 
+import androidx.databinding.ObservableField
 import androidx.lifecycle.ViewModel
+import ip.team13.petowner.data.repository.AuthRepository
+import ip.team13.petowner.helpers.isValidEmail
 
 class LoginViewModel : ViewModel() {
+
+    private val authRepository = AuthRepository()
```

Comment:
We should try to inject the Repositories, rather than initializing them inside each component. Please check the way I did it with the `LeaderboardRepository`.

1. It should be declared as a parameter in all the components, view models needed
2. It should be declared as a `single` in the `AppModule` file (use `get()` there)

Reply:
Changed, thanks